src/lib/codec-wrappers/codec.ts

```diff
@@ -1,5 +1,5 @@
 export interface Encoder {
-  encode(data: ImageData): Promise<ArrayBuffer>;
+  encode(data: ImageData): Promise<ArrayBuffer | SharedArrayBuffer>;
```

Comment:
Iiiinteresting. I removed the `SharedArrayBuffer` and it still compiled. 

I just cleaned the entire repo from build artifacts and now I am seeing the same error. I guess we have some caching in the build system somewhere.

Reply:
On second thought: I’d actually prefer to enforce `ArrayBuffer` as wasm cannot run on `SharedArrayBuffer`, so it’s not really a possibility anyways.

@jakearchibald: I now remember why this was a thing before: The wasm/Emscripten modules export `HEAP8`, `HEAP16` etc as views onto the memory, which are `Int8Array`, `Int16Array` etc. So `HEAP8.buffer` has type `ArrayBuffer | SharedArrayBuffer`. 

I’ll push a commit to this branch.